packages/server/graphql/public/queries/search.ts

```diff
@@ -0,0 +1,382 @@
+    ors.push(eb('m.userId', '=', userId as any))
+
+    // 2. User is in the team
+    if (userTeamIds.length > 0) {
+      ors.push(eb('m.teamId', 'in', userTeamIds))
+    }
+
+    // 3. Page specific sharing
+    ors.push(
+      eb.and([
+        eb('m.objectType', '=', 'page' as any),
+        eb.exists(
+          eb
+            .selectFrom('PageUserAccess as pua')
+            .whereRef('pua.pageId', '=', sql<string>`cast("m"."refId" as integer)`)
+            .where('pua.userId', '=', userId)
+        )
+      ])
+    )
+
+    // 4. Public Meeting Templates
+    ors.push(
+      eb.and([
+        eb('m.objectType', '=', 'meetingTemplate' as any),
+        eb.exists(
+          eb
+            .selectFrom('MeetingTemplate as mt_public')
+            .whereRef('mt_public.id', '=', 'm.refId')
+            .where('mt_public.scope', '=', 'PUBLIC' as any)
+        )
+      ])
+    )
+
+    return eb.and([
+      eb.or(ors),
+      // explicit filters
+      filter?.teamIDs && filter.teamIDs.length > 0
+        ? eb('m.teamId', 'in', filter.teamIDs)
+        : filter?.teamId
+          ? eb('m.teamId', '=', filter.teamId)
+          : eb.val(true),
+      filter?.objectType && filter.objectType.length > 0
+        ? eb('m.objectType', 'in', filter.objectType)
+        : eb.val(true),
+      // date filters
+      filter?.startDate ? eb('m.refUpdatedAt', '>=', filter.startDate) : eb.val(true),
+      filter?.endDate ? eb('m.refUpdatedAt', '<=', filter.endDate) : eb.val(true),
+      // Archived filters (exclude archived by default)
+      eb.or([
+        eb.and([
+          eb('m.objectType', '=', 'page'),
+          eb.exists(
+            eb
+              .selectFrom('Page as p_arch')
+              .whereRef('p_arch.id', '=', sql<number>`cast("m"."refId" as integer)`)
+              .where('p_arch.deletedAt', 'is', null)
+          )
+        ]),
+        eb.and([
+          eb('m.objectType', '=', 'meetingTemplate'),
+          eb.exists(
+            eb
+              .selectFrom('MeetingTemplate as mt_arch')
+              .whereRef('mt_arch.id', '=', 'm.refId')
+              // isActive is boolean, true means active (not archived)
+              .where('mt_arch.isActive', '=', true)
+          )
+        ]),
+        eb.and([eb('m.objectType', '=', 'retrospectiveDiscussionTopic'), eb.val(true)])
+      ])
+    ])
+  }
+
+  // Define search limit (fetch more for RRF)
+  const limit = rawLimit || 20
+  const searchLimit = limit * 3
+
+  // Lexical Search on CHUNKS
+  const lexicalResults = await pg
+    .selectFrom('Embeddings_ember_1 as e')
+    .innerJoin('EmbeddingsMetadata as m', 'e.embeddingsMetadataId', 'm.id')
+    .select([
+      'm.id as metadataId',
+      'e.id as chunkId',
+      sql<number>`ts_rank_cd(e."tsv", plainto_tsquery('english', ${query}), 32)`.as('score'),
+      'e.embedText'
+    ])
+    .where((eb) => buildFilters(eb))
+    .where(sql<boolean>`e."tsv" @@ plainto_tsquery('english', ${query})`)
```

Comment:
+1 `websearch_to_tsquery` may be more appropriate? it'll allow you to quote phrases when 1 word should follow another, etc.